--- /tmp/texture.h.orig	2026-07-13 19:37:05.873370873 -0600
+++ /tmp/texture.h.patched	2026-07-13 19:38:19.365187654 -0600
@@ -669,7 +669,7 @@
 #if defined(__CUDA_ARCH__)
         return tex2D<float>(wp_texture_object(tex.tex), u, v);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return 0.0f;
+        return tex2D<float>(wp_texture_object(tex.tex), u, v);
 #else
         if (tex.tex == 0)
             return 0.0f;
@@ -683,7 +683,7 @@
 #if defined(__CUDA_ARCH__)
         return tex3D<float>(wp_texture_object(tex.tex), u, v, w);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return 0.0f;
+        return tex3D<float>(wp_texture_object(tex.tex), u, v, w);
 #else
         if (tex.tex == 0)
             return 0.0f;
@@ -715,7 +715,8 @@
         float2 val = tex2D<float2>(wp_texture_object(tex.tex), u, v);
         return vec2f(val.x, val.y);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return vec2f(0.0f, 0.0f);
+        float2 val = tex2D<float2>(wp_texture_object(tex.tex), u, v);
+        return vec2f(val.x, val.y);
 #else
         if (tex.tex == 0)
             return vec2f(0.0f, 0.0f);
@@ -730,7 +731,8 @@
         float2 val = tex3D<float2>(wp_texture_object(tex.tex), u, v, w);
         return vec2f(val.x, val.y);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return vec2f(0.0f, 0.0f);
+        float2 val = tex3D<float2>(wp_texture_object(tex.tex), u, v, w);
+        return vec2f(val.x, val.y);
 #else
         if (tex.tex == 0)
             return vec2f(0.0f, 0.0f);
@@ -765,7 +767,8 @@
         float4 val = tex2D<float4>(wp_texture_object(tex.tex), u, v);
         return vec4f(val.x, val.y, val.z, val.w);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return vec4f(0.0f, 0.0f, 0.0f, 0.0f);
+        float4 val = tex2D<float4>(wp_texture_object(tex.tex), u, v);
+        return vec4f(val.x, val.y, val.z, val.w);
 #else
         if (tex.tex == 0)
             return vec4f(0.0f, 0.0f, 0.0f, 0.0f);
@@ -783,7 +786,8 @@
         float4 val = tex3D<float4>(wp_texture_object(tex.tex), u, v, w);
         return vec4f(val.x, val.y, val.z, val.w);
 #elif defined(__HIP_DEVICE_COMPILE__)
-        return vec4f(0.0f, 0.0f, 0.0f, 0.0f);
+        float4 val = tex3D<float4>(wp_texture_object(tex.tex), u, v, w);
+        return vec4f(val.x, val.y, val.z, val.w);
 #else
         if (tex.tex == 0)
             return vec4f(0.0f, 0.0f, 0.0f, 0.0f);
